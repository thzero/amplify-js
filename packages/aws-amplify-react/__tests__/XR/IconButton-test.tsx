import * as React from 'react';
import IconButton from '../../src/XR/IconButton';

describe('IconButton', () => {
	test('renders successfully with no variant', () => {
		const wrapper = mount(<IconButton />);
		expect(wrapper).toMatchSnapshot();
	});

	test('renders successfully with variant', () => {
		const wrapper = mount(<IconButton variant="maximize" />);
		expect(wrapper).toMatchSnapshot();
	});
});
