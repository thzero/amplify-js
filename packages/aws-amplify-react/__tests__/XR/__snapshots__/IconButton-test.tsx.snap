// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders successfully with no variant 1`] = `
<IconButton>
  <Tooltip>
    <div
      className="XR__tooltip___UYyhn"
    >
      <button
        className="XR__actionButton___2poIM"
      />
    </div>
  </Tooltip>
</IconButton>
`;

exports[`IconButton renders successfully with variant 1`] = `
<IconButton
  variant="maximize"
>
  <Tooltip>
    <div
      className="XR__tooltip___UYyhn"
    >
      <button
        className="XR__actionButton___2poIM"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fillRule="evenodd"
            id="icons/minis/screenfull"
          >
            <path
              d="M2.04162598,3 L2.04162598,16 L17.0147705,16 L17.0147705,3 L2.04162598,3 Z M1,2 L18,2 L18,17 L1,17 L1,2 Z M3,4 L16,4 L16,15 L3,15 L3,4 Z"
              fill="#FFF"
              fillRule="nonzero"
              id="Rectangle-Copy"
            />
          </g>
        </svg>
      </button>
    </div>
  </Tooltip>
</IconButton>
`;
